pkg/kev/config/schema.go

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2020 Appvia Ltd <info@appvia.io>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package config
+
+var ServicesSchema = map[string]interface{}{
+	"$schema": "http://json-schema.org/draft/2019-09/schema#",
+	"$id":     "http://appvia.io/schemas/kev-service-labels-schema.json",
+	"type":    "object",
+	"properties": map[string]interface{}{
+		LabelComponentEnabled:                  map[string]interface{}{"type": "string", "enum": []string{"false", "true"}},
+		LabelServiceType:                       map[string]interface{}{"type": "string", "enum": []string{"None", "Headless", "ClusterIP", "Nodeport", "LoadBalancer"}},
+		LabelWorkloadCPU:                       map[string]interface{}{"type": "string"},
+		LabelWorkloadImagePullPolicy:           map[string]interface{}{"type": "string", "enum": []string{"IfNotPresent", "Always"}},
+		LabelWorkloadLivenessProbeCommand:      map[string]interface{}{"type": "string"},
+		LabelWorkloadLivenessProbeDisabled:     map[string]interface{}{"type": "string", "enum": []string{"false", "true"}},
+		LabelWorkloadLivenessProbeInitialDelay: map[string]interface{}{"type": "string", "format": "duration"},
+		LabelWorkloadLivenessProbeInterval:     map[string]interface{}{"type": "string", "format": "duration"},
+		LabelWorkloadLivenessProbeRetries:      map[string]interface{}{"type": "string", "pattern": `^\d+$`},
+		LabelWorkloadLivenessProbeTimeout:      map[string]interface{}{"type": "string", "format": "duration"},
+		LabelWorkloadMaxCPU:                    map[string]interface{}{"type": "string"},
+		LabelWorkloadMaxMemory:                 map[string]interface{}{"type": "string", "pattern": `^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$`},
```

Comment:
I guess we shouldn't allow P :) T is also questionable. Maybe we should allow up to G? Is this regex going to cover cases like `Gi` or `Mi`? Same comment applies to the line below.

Reply:
This regex is quite permissive - lifted from K8s.
- Happy to remove P & T.
- And, yes it handles `Mi` and `Gi`.

Reply:
Technically it's OK, so could leave as is and if someone requests a few Petabytes of memory it's unlikely to fit on any host anyway...

Reply:
@marcin Rolled back update and left the regex as is.